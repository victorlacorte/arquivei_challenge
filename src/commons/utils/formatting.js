function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
