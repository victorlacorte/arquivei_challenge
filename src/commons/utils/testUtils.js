/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
