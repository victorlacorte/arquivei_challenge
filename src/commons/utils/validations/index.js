export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
