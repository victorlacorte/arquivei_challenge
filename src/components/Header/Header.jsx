import React, { memo } from 'react';
import styled from 'styled-components';

const StyledHeader = styled.header`
  position: sticky;
  top: 0;
  background-color: ${({ theme }) => theme.palette.primary.main};
  height: 60px;
  padding: 16px;
  z-index: ${({ theme }) => theme.zIndex.header};

  ${({ theme }) => theme.breakpoints.down('sm')`
    height: 40px;
    padding: 8px;
  `}
`;

const StyledImg = styled.img`
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  width: 30%;
  max-width: 352px;
  height: auto;

  ${({ theme }) => theme.breakpoints.down('sm')`
    width: 50%;
  `}
`;

const Header = () => (
  <StyledHeader>
    <StyledImg src="/images/arquivei-lite.gif" alt="Arquivei Lite logo" />
  </StyledHeader>
);

export default memo(Header);
