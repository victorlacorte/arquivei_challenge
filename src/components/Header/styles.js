import styled from 'styled-components';

/**
 * Total header height:
 * Desktop: 60 + 16 * 2 == 92px
 * Mobile: 40 + 8 * 2 = 56px
 *
 * This information could be in the theme since our Main component makes use of it
 */
const StyledHeader = styled.header`
  position: sticky;
  top: 0;
  background-color: ${({ theme }) => theme.palette.primary.main};
  box-shadow: 0 3px 5px ${({ theme }) => theme.palette.grey400.main};
  height: 60px;
  padding: 16px;
  z-index: ${({ theme }) => theme.zIndex.appBar};

  ${({ theme }) => theme.breakpoints.down('sm')`
    height: 40px;
    padding: 8px;
  `}
`;

const StyledImg = styled.img`
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  width: 30%;
  max-width: 352px;
  height: auto;
  cursor: pointer;

  ${({ theme }) => theme.breakpoints.down('sm')`
    width: 50%;
  `}
`;

export {
  StyledHeader,
  StyledImg,
};
