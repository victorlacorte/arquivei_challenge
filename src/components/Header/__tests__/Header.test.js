import React from 'react';

import Header from 'components/Header';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<Header />', () => {
  it('creates a Header component', () => {
    const wrapper = shallowWithTheme(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
