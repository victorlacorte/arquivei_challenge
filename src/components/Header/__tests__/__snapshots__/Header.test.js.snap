// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> creates a Header component 1`] = `
<styles__StyledHeader>
  <styles__StyledImg
    alt="Arquivei Lite logo"
    onClick={[Function]}
    src="/images/arquivei-lite.gif"
  />
</styles__StyledHeader>
`;
